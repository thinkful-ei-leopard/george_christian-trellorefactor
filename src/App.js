import React, { Component } from 'react';
import List from './List'
import './App.css';

class App extends Component {
  static defaultProps = {
    store: {
      lists: [],
      allCards: {},
    }
  };

  state = {
     lists: [
      {
        id: '1',
        header: 'First list',
        cardIds: [ 'a', 'b', 'e', 'f', 'g', 'j', 'l', 'm' ],
      },
      {
        id: '2',
        header: 'Second list',
        cardIds: ['b', 'c', 'd', 'f', 'h', 'i', 'k'],
      },
      {
        id: '3',
        header: 'Third list',
        cardIds: [ 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm' ],
      },
      {
        id: '4',
        header: 'Fourth list',
        cardIds: [ 'l', 'm' ],
      },
    ],
    allCards: {
      'a': { id: 'a', title: 'First card', content: 'lorem ipsum' },
      'b': { id: 'b', title: 'Second card', content: 'lorem ipsum' },
      'c': { id: 'c', title: 'Third card', content: 'lorem ipsum' },
      'd': { id: 'd', title: 'Fourth card', content: 'lorem ipsum' },
      'e': { id: 'e', title: 'Fifth card', content: 'lorem ipsum' },
      'f': { id: 'f', title: 'Sixth card', content: 'lorem ipsum' },
      'g': { id: 'g', title: 'Seventh card', content: 'lorem ipsum' },
      'h': { id: 'h', title: 'Eighth card', content: 'lorem ipsum' },
      'i': { id: 'i', title: 'Ninth card', content: 'lorem ipsum' },
      'j': { id: 'j', title: 'Tenth card', content: 'lorem ipsum' },
      'k': { id: 'k', title: 'Eleventh card', content: 'lorem ipsum' },
      'l': { id: 'l', title: 'Twelfth card', content: 'lorem ipsum' },
      'm': { id: 'm', title: 'Thirteenth card', content: 'lorem ipsum' },
    },
  }

newRandomCard = listId => {
    console.log('clicked');
    const id = Math.random().toString(36).substring(2, 4)
      + Math.random().toString(36).substring(2, 4);
    console.log(id)
    const newCard = {
      id,
      title: `Random Card ${id}`,
      content: 'The New Card',
    }
    //added to cardId's array under current list
    let indexOf = this.state.lists.findIndex(x => x.id === listId);
    let lists = this.state.lists;
    console.log(listId)
    lists[indexOf].cardIds = [...this.state.lists[indexOf].cardIds, id]
    this.setState({lists});
   



    //add to allCards object
    let newCardObject = {id: newCard.id, title: newCard.title, content: newCard.content}
    let allCards = this.state.allCards;
    allCards[newCardObject.id] = newCardObject;
    this.setState({allCards});

  }
  

  render() {
    console.log(this.state.lists.cardIds)
    // const { store } = this.props
    return (
      <main className='App'>
        <header className='App-header'>
          <h1>Trelloyes!</h1>
        </header>
        <div className='App-list'>
          {this.state.lists.map(list => (
            <List
              key={list.id}
              listId={list.id}
              header={list.header}
              cards={list.cardIds.map(id => this.state.allCards[id])}
              makeNewRandomCard={this.newRandomCard}
            />
          ))}
        </div>
      </main>
    );
  }
}


export default App;