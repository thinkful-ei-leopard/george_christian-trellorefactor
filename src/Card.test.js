import React from 'react'
import Card from './Card'
import renderer from 'react-test-renderer'
import App from './App'
import ReactDOM from 'react-dom'

describe('suite tests', () =>{

    it('renders', () => {
        const tree = renderer.create(<Card />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Card />, div);
        ReactDOM.unmountComponentAtNode(div);
    })

});


