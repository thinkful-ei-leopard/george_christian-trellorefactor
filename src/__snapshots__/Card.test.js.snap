// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3 />
  <p />
</div>
`;

exports[`suite tests renders 1`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3 />
  <p />
</div>
`;
